erts/emulator/beam/erl_bif_binary.c

```diff
@@ -2457,6 +2459,55 @@ static int cleanup_copy_bin_state(Binary *bp)
     return 1;
 }
 
+static BIF_RETTYPE binary_linear_compare(Process *p, Eterm bin1, Eterm bin2)
+{
+    byte *bytes1;
+    byte *bytes2;
+    Uint bitoffs1, bitsize1;
+    Uint bitoffs2, bitsize2;
+    Uint size;
+    size_t i;
+    volatile unsigned char x = 0U;
+
+    if (is_not_binary(bin1) || is_not_binary(bin2))
+        goto bad_arg;
+
+    size = binary_size(bin1);
+
+    if (size != binary_size(bin2))
+        goto bad_arg;
+    
+    ERTS_GET_BINARY_BYTES(bin1, bytes1, bitoffs1, bitsize1); 
+    ERTS_GET_BINARY_BYTES(bin2, bytes2, bitoffs2, bitsize2);
+    
+    const volatile unsigned char *volatile b1 =
```

Comment:
@jhogberg @IngelaAndin This is a best effort to prevent compiler optimizations. It makes sense to me, but it could also be seen as a little too hand wavey. Thoughts? 

Note I did not do the weak dummy function hack as done in libsodium as that seemed to me as a form of mitigation for a very specific case that does not apply to us (i.e., the case where another lib in C has hard coded values it checks against).

Reply:
The commit message hints at there being a good reason for this by pointing to two dead links. :(

https://github.com/jedisct1/libsodium/commit/113091b2a0fbc999f6ef585c10bd622e94f8b6eb